Implement public API for high-DPI

kryptan opened the pull request:
As discussed in #105. This pull request contains only changes to the winit API, no implementation in any of the backends yet.

1. Add `MonitorId::get_hidpi_factor() -> f32` method. Currently returns 1 for all backends.
2. Add `WindowEvent::HiDPIFactorChanged(f32)`.
3. Remove `Window::get_inner_size_points` and  `Window::get_inner_size_pixels` .

vberger commented on src/events.rs:
```diff
@@ -87,6 +87,17 @@ pub enum WindowEvent {
+    /// * A user changes the desktop scaling value (e.g. in Control Panel on Windows).
+    /// * A user moves the application window to a display with a different DPI. This message is sent when a window’s
+    /// position changes such that most of its area lies on a monitor with a DPI that is different from the DPI
+    /// before the position change.
```
Actually this behavior will be platform dependent.

For example, on wayland, we don't have the information of how much the window covers a monitor. We just have the list of all monitors that display some part of the surface. Thus, the wayland backend will just return the highest dpi factor of all the monitors currently displaying some part of the window.

kryptan replied:
I removed that part

tomaka commented on src/window.rs:
```diff
@@ -183,54 +183,25 @@ impl Window {
-    ///
-    /// Returns `None` if the window no longer exists.
-    #[inline]
-    pub fn get_inner_size_points(&self) -> Option<(u32, u32)> {
```
These two functions should be deprecated, not removed instantly.

kryptan commented:
@tomaka  `get_inner_size_points` and `get_inner_size_pixels` assumed that `get_inner_size` returns points. This was incorrect at least on Windows and if DPI-awareness was enabled `get_inner_size_pixels` would return incorrect size on high-DPI display. I changed their implementations so that they are now assuming that `get_inner_size` return pixels. I checked backends and X11 was returning points from `get_inner_size` by manually converting pixels to points, I fixed that. All other backends seem to return size directly from OS functions.

tomaka commented on src/window.rs:
```diff
@@ -202,6 +202,36 @@ impl Window {
+    /// Returns `None` if the window no longer exists.
+    ///
+    /// DEPRECATED
+    #[inline]
```
I didn't mention it, but it should have a `#[deprecated]` attribute so that a compiler warning is shown when you use them.

kryptan replied:
added

tomaka commented:
Err, sorry for the ping-pong, but the last thing that's missing is an entry in `CHANGELOG.md`.

kryptan commented:
I updated changelog. Is there anything else that's need to be done?

tomaka commented:
Thanks
